### PR TITLE
[persistence] Use Java 17 features

### PR DESCRIPTION
**Pull request by holgerfriedrich**
- Use instanceof matching and multiline strings.
- Use Map/Set/List.of instead of Collections.

Similar to what we did in core a few month ago. Note that this breaks compatibilty to Java 11.

### CODE REVIEW

**Comment by jlaur** [comment]
Just a heads-up, there are some conflicts with #15434.

**Comment by holgerfriedrich** [comment]
@jlaur I excluded ```JdbcBaseDAO.java``` from my PR, should not cause a conflict anymore

File: bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java [hunk 1]
```diff
@@ -302,17 +302,17 @@ public static DynamoDBItem<?> fromStateNew(Item item, ZonedDateTime time, @Nulla
...
                     case PLAY:
                         return new DynamoDBBigDecimalItem(name, PLAY_BIGDECIMAL, time, expireDays);
                     case PAUSE:
                         return new DynamoDBBigDecimalItem(name, PAUSE_BIGDECIMAL, time, expireDays);
                     default:
                         throw new IllegalArgumentException("Unexpected enum with PlayPauseType: " + state.toString());
                 }
-            } else if (state instanceof RewindFastforwardType) {
-                switch ((RewindFastforwardType) state) {
+            } else if (state instanceof RewindFastforwardType type) {
```
Comment (lsiepel) [inline bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java:314]: I see it here and elsewhere. The generated instanceof patterns do not create a meaningfull/logical variable name. 
like 
… instanceof RewindFastforwardType type

Not wrong, but consider more specific bar names.

Comment (holgerfriedrich) [reply bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java:314]: Completely agreed....

File: bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/DynamoDBConfig.java [hunk 2]
```diff
@@ -93,8 +93,10 @@ public class DynamoDBConfig {
                 String profile = (String) config.get("profile");
                 if (profilesConfigFile == null || profilesConfigFile.isBlank() || profile == null
                         || profile.isBlank()) {
-                    LOGGER.error("Specify either 1) accessKey and secretKey; or 2) profilesConfigFile and "
-                            + "profile for providing AWS credentials");
+                    LOGGER.error("""
+                            Specify either 1) accessKey and secretKey; or 2) profilesConfigFile and \
+                            profile for providing AWS credentials\
```
Comment (ssalonen) [inline bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/DynamoDBConfig.java:98]: Is there some extra space in here that would end up in being printed as well?

File: bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java [hunk 3]
```diff
@@ -411,8 +410,8 @@ public static DynamoDBItem<?> fromStateNew(Item item, ZonedDateTime time, @Nulla
                     if (numberState == null) {
                         return null;
                     }
-                    if (item instanceof NumberItem) {
-                        NumberItem numberItem = ((NumberItem) item);
+                    if (item instanceof NumberItem numberItem1) {
+                        NumberItem numberItem = numberItem1;
```
Comment (ssalonen) [inline bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java:414]: Is this line needed?

File: bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/DynamoDBConfig.java (hunk 2 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/DynamoDBConfig.java:98]: Just checked, it is fine. The outputs of both commands look similar (except the number I added at the end)
```
[ERROR] [...] - Specify either 1) accessKey and secretKey; or 2) profilesConfigFile and profile for providing AWS credentials1
[ERROR] [...] - Specify either 1) accessKey and secretKey; or 2) profilesConfigFile and profile for providing AWS credentials2
````

File: bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java (hunk 3 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java:414]: fixed

Comment (holgerfriedrich) [reply bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java:414]: fixed

**Review by lsiepel — APPROVED** [review]
LGTM

**Review by wborn — APPROVED** [review]
LGTM!

**Pull request merged**